default-enarx-trust-process.md

```diff
@@ -0,0 +1,474 @@
+state 1 to state 2.  It is at this point that at **TEE instance** on the
+host (and in the **host owner's trust domain**), loaded with an **Enarx
+runtime image** (which then executes, to become an **Enarx runtime
+instance**) becomes an **Empty Keep**, within the **tenant trust domain**
+(having moved out of the **host owner domain**. This is achieved by the
+use of a __trust pivot__: the **host CPU + CPU firmware**.  The attestation
+process described in the State 1->2 change section explicitly notes that
+the cryptographic identity of the **host CPU + CPU firmware** must be
+validated as part of the attestation measurement check process.
+
+The ability for the tenant to trust the host CPU + CPU firmware underpins
+the entire trust model process, and it is vital that all implementations
+check it very carefully.
+
+### State 0->1 change
+
+#### State 0 - initial state
+
+ - host owner trust domain
+    - host
+    - host CPU + CPU firmware
+    - Enarx host agent
+    - Enarx runtime image
+    - other workloads
+ - tenant trust domain
+    - Orchestrator
+    - Enarx client agent
+    - tenant workload image
+    - tenant workload image repository
+
+#### State 1
+
+ - host owner trust domain
+    - host
+    - host CPU + CPU firmware
+    - Enarx host agent
+    - **TEE instance**
+    - **Enarx runtime instance (within TEE instance)**
+    - other workloads
+ - tenant trust domain
+    - Orchestrator
+    - Enarx client agent
+    - tenant workload image
+    - tenant workload image repository
+
+#### Changes
+
+The changes between state 0 and state 1 are the creation of a TEE instance
+and the loading of the Enarx runtime image into that instance.
+
+Note that the **Enarx host agent**, which is considered as a single component
+in this document for the purposes of trust domain discusions, can be further
+decomposed into the following components:
+ - Keep manager
+   - one per host (future trust models might allow more than one per host)
+   - keeps track of all Keeps on the host
+   - launches Keep loader components (see below)
+   - may persist state to allow recovery, or poll existing Keeps if required
+ - Keep loader
+   - one per Keep
+   - lifespan coterminous with the Keep it manages
+   - causes the creation of the **TEE instance**
+   - loads **Enarx runtime instance** into **TEE instance**
+   - creates a Main loop component
+ - Main loop
+   - one per Keep
+   - lifespan cotermnous with the Keep it services
+   - provides the syscall processing from outside the Keep
+
+The initialisation of the process is prompted by the **Orchestrator**, which
+communicates with the **Enarx client agent**, which contacts the **Enarx
+host agent** over a network connection (which does not need to be
+encrypted, as no confidential data is passed across it, because the
+**Enarx host agent** is not a trusted entity - as it is part of the **tenant
+host domain** - anyway. A mechanism (such as a cryptographic nonce), allowing
```

Comment:
```suggestion
host agent** over a network connection (which does not need to be
encrypted, as no confidential data is passed across it, because the
**Enarx host agent** is not a trusted entity - as it is part of the **tenant
host domain**). A mechanism (such as a cryptographic nonce), allowing
```